e2e/bitrise.yml

```
@@ -0,0 +1,95 @@
+format_version: "11"
+default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
+
+app:
+  envs:
+  # Shared secrets for testing, use a .bitrise.secrets.yml file to define these locally
+  - BITRISEIO_CACHE_SERVICE_URL: $BITRISEIO_CACHE_SERVICE_URL
+  - CACHE_API_CLIENT_SECRET: $CACHE_API_CLIENT_SECRET
+
+workflows:
+  test_gradle_output_diagnostics:
+    envs:
+    - TEST_APP_URL: https://github.com/bitrise-io/Bitrise-Android-Sample
+    - BRANCH: main
+    before_run:
+    - _generate_api_token
+    - _setup
+    steps:
+    - change-workdir:
+        title: Switch working dir to _tmp
+        inputs:
+        - path: ./_tmp
+    # - set-java-version@1:
+    - android-build:
+        title: Build APK
+        inputs:
+        - module: app
+        - variant: debug
+    - script:
+        title: Save cache
+        inputs:
+        - content: |-
+            set -ex
+            ../bitrise-build-cache-cli -d save-gradle-output-data
+    - script:
+        title: Delete local Gradle caches
+        inputs:
+        - content: |-
+            set -ex
+            sudo rm -rf .gradle
+    - script:
+        title: Restore cache
+        inputs:
+        - content: |-
+            set -ex
+            ../bitrise-build-cache-cli -d restore-gradle-output-data
+    - script:
+        title: Verify restored folder
+        inputs:
+        - content: |-
+            set -ex
+            if [ ! -d .gradle/caches ]; then
```

```suggestion
            if [ ! "$(ls -A .gradle)" ]; then
```

Gradle stores the local cache in `$USER_HOME/.gradle/caches` by default (and not in the project's `.gradle` folder) so this will always fail. Verifying that the folder is not empty should work.